rfc/cart-api-v1.md

```diff
@@ -0,0 +1,251 @@
+# Version 1
+
+| Issue | Status |
+| ----- | ------ |
+| #88   | Draft  |
+
+## Data Models
+
+There are two data models required for Cart API: `Cart` and `CartItem`. They are
+designed in the way that allows multiple types of entity in a single `Cart`.
+
+### `Cart`
+| Property | Type | Description | Note |
+| --- | --- | --- | --- |
+| `id` | UUID | Cart ID, auto-generated by the web API | - |
+| `user_id` | UUID | The ID of the user that creates the collection | - |
+| `cart_name` | `str` | Cart Name: If it is `NULL`, the cart is the default one for a corresponding user. | **NEED MORE DISCUSSION** |
+| `default_cart` | `bool` | Flag to indicate whether this cart is the default one for a corresponding user (`False` by default) | **NEED MORE DISCUSSION** |
+
+For the details on the default cart, see `GET /resources/carts/{id}`.
+
+### `CartItem`
+| Property | Type | Description |
+| --- | --- | --- |
+| `id` | `str` | Cart Item ID, hashed by the combination of `` |
+| `cart_id` | UUID | Associated `Cart.id` |
+| `entity_id` | UUID | Entity ID |
+| `entity_type` | `str` | Entity Type |
+| `bundle_uuid` | UUID | Bundle ID |
+| `bundle_version` | `str` | Bundle Version |
+
+> Cart Item ID is a hash string of the combination of `cart_id`, `entity_id`, `bundle_uuid`, `bundle_version`. The hash algorithm is probably `sha256` but it is not finalized.
+
+### `CartItemBatchJob`
```

Comment:
May be useful to add a cart id here so we can query for in progress jobs for a given cart.  But this is dependent on how the frontend is designed.